sqld/src/hrana/batch.rs

```diff
@@ -0,0 +1,89 @@
+use anyhow::Result;
+
+use crate::database::{Cond, Database, Program, Step};
+
+use super::proto;
+use super::stmt::{
+    proto_error_from_stmt_error, proto_stmt_result_from_query_response, proto_stmt_to_query,
+    stmt_error_from_sqld_error,
+};
+
+#[derive(thiserror::Error, Debug)]
+pub enum BatchError {
+    #[error("Invalid reference to step in a condition")]
+    CondBadStep,
+}
+
+fn proto_cond_to_cond(cond: &proto::BatchCond) -> Result<Cond> {
+    let try_convert_step = |step: i32| -> std::result::Result<usize, BatchError> {
```

Comment:
When we are already checking the step index, perhaps we can also check that it refers to step that is before the current step?

Reply:
this is done when the program is executed there: https://github.com/libsql/sqld/pull/307/files#diff-b7c90481e4da77d124d59b67dd5c2141bfba58deac7df43808defee65815fef9R308